## Iris Data Set - Project 2019                                  
### Programing and Scripting 
### GMIT Galway
Slawomir Sowa
12/04/2019

### Subject

Project concern Fisher’s Iris data set. 

1. Research background information about the data set and write a summary about it.
2. Keep a list of references you used in completing the project.
3. Download the data set and write some Python code to investigate it.
4. Summarise the data set by, for example, calculating the maximum, minimum and mean of each column of the data set. A Python script will quickly do this for you.
5. Write a summary of your investigations.
6. Include supporting tables and graphics as you deem necessary.

<img width='400px' align='left' src='http://miriamposner.com/classes/dh101f16/wp-content/uploads/sites/5/2016/11/Python_logo.png'>




### Fisher's Iris Data Set
(history information about data-set to insert here)

To read data from `iris.csv` file I use `pandas` library and `read_csv` function. Variable `data` is my data frame. `data.head(5)` allow mi to print first 5 rows of data set.    

In [1]:
import pandas as pd

data = pd.read_csv('iris.csv')
data.head(5)

Id  SepalLengthCm  SepalWidthCm  PetalLengthCm  PetalWidthCm      Species
0   1            5.1           3.5            1.4           0.2  Iris-setosa
1   2            4.9           3.0            1.4           0.2  Iris-setosa
2   3            4.7           3.2            1.3           0.2  Iris-setosa
3   4            4.6           3.1            1.5           0.2  Iris-setosa
4   5            5.0           3.6            1.4           0.2  Iris-setosa

As we see above, our data set contains 6 columns: `ID`, `SepalLengthCm`, `SepalWidthCm`, `PetalLengthCm`,`PetalWidthCm` and `Species`. 

Firstly I explaine what Sepal and Petal Length are. 

<img src="/files/sepalpetal.jpg" width="227" />

As we see on photo above, Sepal  is a longer part of iris flower. Sepal Width and Sepal Length are measured in cm. 
Petal is shorther part of iris flower. Petal width and Petal Length are measured in cm.  

In [2]:
data.tail(5)

Id  SepalLengthCm  SepalWidthCm  PetalLengthCm  PetalWidthCm  \
145  146            6.7           3.0            5.2           2.3   
146  147            6.3           2.5            5.0           1.9   
147  148            6.5           3.0            5.2           2.0   
148  149            6.2           3.4            5.4           2.3   
149  150            5.9           3.0            5.1           1.8   

            Species  
145  Iris-virginica  
146  Iris-virginica  
147  Iris-virginica  
148  Iris-virginica  
149  Iris-virginica

`data.tail(5)` function gives me last 5 rows taken from data set. As we see there is 150 rows. `ID` column numbered all rows. 

Interesting things happen in `species` column. We started with `iris-setosa` and finished with `iris-virginica`. Lets check how many different species we have in data set. 

In [3]:
data['Species'].value_counts()

Iris-setosa        50
Iris-virginica     50
Iris-versicolor    50
Name: Species, dtype: int64

By using `value_counts()` function on my data frame `data` I found 3 unique values in `Spacies` column:`Iris-setosa`, `Iris-virginica` and `Iris-versicolor` and each type occurs 50 times.    

### Iris-setosa Analyse

Now let's take a closer look on Iris-setosa type. Firstly I have to retrive iris-setosa species from data set and save it in to separate data frame `dfSetosa`.  

In [12]:
dfSetosa = data[data['Species'] == 'Iris-setosa']
dfSetosa['Species'].value_counts()


Iris-setosa    50
Name: Species, dtype: int64

`dfSetosa['Species'].value_counts()` shows that new dataframe has 50 rows and contains only Iris-Setosa type. 
Now I calucate max, min, mean and standard deviation value for Sepal and Petal length and width.

In [13]:
dfSetosa = dfSetosa.drop(['Id'], axis = 1)
dfSetosa.describe()

SepalLengthCm  SepalWidthCm  PetalLengthCm  PetalWidthCm
count       50.00000     50.000000      50.000000      50.00000
mean         5.00600      3.418000       1.464000       0.24400
std          0.35249      0.381024       0.173511       0.10721
min          4.30000      2.300000       1.000000       0.10000
25%          4.80000      3.125000       1.400000       0.20000
50%          5.00000      3.400000       1.500000       0.20000
75%          5.20000      3.675000       1.575000       0.30000
max          5.80000      4.400000       1.900000       0.60000

For better view i decided to drop `Id` column by using `DataFrame.drop()` function.
As we see above 